eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/log/EventLog.scala

```diff
@@ -98,33 +106,43 @@ case class BatchReadResult(events: Seq[DurableEvent], to: Long) extends DurableE
 case class DeletionMetadata(toSequenceNr: Long, remoteLogIds: Set[String])
 
 /**
+ * Result of an event batch-read operation.
+ *
+ * @param events Read event batch.
+ * @param to Last read position in the event log.
+ */
+case class BatchReadResult(events: Seq[DurableEvent], to: Long) extends DurableEventBatch
+
+/**
  * Indicates that a storage backend doesn't support physical deletion of events.
  */
 private class PhysicalDeletionNotSupportedException extends UnsupportedOperationException
 
 /**
  * Event log storage provider interface (SPI).
+ *
+ * @tparam A Event log state type.
  */
-trait EventLogSPI { this: Actor =>
+trait EventLogSPI[A] { this: Actor =>
```

Comment:
That's actually only a constraint of the abstract `EventLog` (from which all implementations extend).